src/ngx_http_lua_common.h

```diff
@@ -176,6 +178,9 @@ struct ngx_http_lua_main_conf_s {
 
     ngx_array_t         *preload_hooks; /* of ngx_http_lua_preload_hook_t */
 
+    ngx_array_t         *delay_list; /* of void* */
+    unsigned             delay_list_inited;
```

Comment:
@rainingmaster I think you should add these as code comments to improve code readability. Also, I think we definitely need better names for these fields.

I don't think we should count the number of registrations. We only need to know whether or not to delay. Your registration API does not really register any real data from the 3rd-party modules. No callbacks from the 3rd-party modules are registered here anyway.